src/partition/repart.c

```diff
@@ -2091,6 +2091,11 @@ static int partition_finalize_fstype(Partition *p, const char *path) {
         return free_and_strdup_warn(&p->format, v);
 }
 
+static bool partition_needs_populate(const Partition *p) {
+        assert(p);
+        return !strv_isempty(p->copy_files) || !strv_isempty(p->make_directories);
```

Comment:
why is p->subvolumes not checked here?

Reply:
Because `Subvolumes=` does not create inodes, it only turns existing inodes into subvolumes